### PR TITLE
Added FSMNLP 2019 for release on 09-22

### PR DESCRIPTION
**Pull request by mjpost**

### CODE REVIEW

File: bin/anthology/sigs.py [hunk 1]
```diff
@@ -36,9 +37,7 @@
 
 def _sigevent_to_repr(event):
     if event.anthology_id is not None:
-        # For some reason, SIG files point to the front matter, not to the
-        # containing proceedings volume -- change that here
-        return to_volume_id(event.anthology_id)
+        return event.anthology_id
     return {"name": event.name, "url": event.url}
```
Comment (mjpost) [inline bin/anthology/sigs.py]: SIG events used to list the frontmatter paper (e.g., `S19-1000`), but this was changed a while ago to list the volume directly.

File: bin/anthology/utils.py [hunk 2]
```diff
@@ -143,6 +137,25 @@ def infer_attachment_url(filename, parent_id=None):
...
+    digits = collection_id[1:]
+    if int(digits) >= 60:
+        year = "19{}".format(digits)
+    else:
+        year = "20{}".format(digits)
+
+    return year
+
+
 _MONTH_TO_NUM = {
```
Comment (mjpost) [inline bin/anthology/utils.py]: Moved here from another file since it is used in multiple places now.

File: bin/anthology/volumes.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 from .papers import Paper
 from .venues import VenueIndex
 from .sigs import SIGIndex
-from .utils import parse_element, is_journal, month_str2num, infer_url
+from anthology.utils import parse_element, is_journal, month_str2num, infer_url, infer_year
```
Comment (mjpost) [inline bin/anthology/volumes.py]: This is the Python 3.7 format for including local packages.

File: bin/anthology/volumes.py [hunk 4]
```diff
@@ -115,23 +116,6 @@ def _set_meta_info(self, meta_data):
...
-            len(self.collection_id) == 3
-        ), "Couldn't infer year: unknown volume ID format"
-        digits = self.collection_id[1:]
-        if int(digits) >= 60:
-            year = "19{}".format(digits)
-        else:
-            year = "20{}".format(digits)
-        self.attrib["year"] = year
-
     @property
```
Comment (mjpost) [inline bin/anthology/volumes.py]: Moved to `utils`.

File: data/xml/W19.xml [hunk 5]
```diff
@@ -3807,6 +3807,144 @@
       <doi>10.18653/v1/W19-3025</doi>
     </paper>
   </volume>
+  <volume id="31" ingest-date="2019-09-22">
+    <meta>
```
Comment (mjpost) [inline data/xml/W19.xml]: Ingest date in the future, which is causing this ruckus with the SIG.

**Comment by mjpost** [comment]
Does someone have a minute to look at this so I can post it by 9/22? @acl-org/anthology

File: bin/requirements.txt [hunk 6]
```diff
@@ -7,3 +7,4 @@ PyYAML>=3.0
 stop-words
 tqdm
 texsoup<=0.1.4
+nltk
```
Comment (akoehn) [inline bin/requirements.txt]: Why do we now depend on nltk?  I couldn't find a matching new import.

File: bin/anthology/sigs.py [hunk 7]
```diff
@@ -56,6 +55,11 @@ def load_from_dir(self, directory):
                 sig = SIG.from_dict(data)
                 self.sigs[sig.acronym] = sig
 
+    def remove_volume(self, full_volume_id):
+        for acronym, sig in self.sigs.items():
+            # print(f'Removing {full_volume_id} from {acronym}')
```
Comment (akoehn) [inline bin/anthology/sigs.py]: I would like to either have a debug call here or the commented out print call completely removed.

File: bin/requirements.txt (hunk 6 above)
Comment (mjpost) [reply bin/requirements.txt]: It's used in `bin/fixedcase/common.py`.

File: bin/anthology/sigs.py (hunk 7 above)
Comment (mjpost) [reply bin/anthology/sigs.py]: Will do!

**Comment by mjpost** [comment]
You set the target date by adding `--ingest-date YYYY-MM-DD` to `./bin/ingest.py`.

**Pull request merged**